bin/bench

```diff
@@ -0,0 +1,278 @@
+#!/usr/bin/env ruby
+require 'fileutils'
+require 'pathname'
+require 'shellwords'
+require 'English'
+require 'net/http'
+require 'json'
+require 'benchmark'
+
+############################
+# USAGE
+#
+# bin/bench <ref1> <ref2>
+#   <ref1> defaults to the current branch
+#   <ref2> defaults to the master branch
+# bin/bench current # will run on the current branch
+# bin/bench revisions 792fb8a9053f8db3c562dae4f40907a582dd1720 master # every revision inclusive
+###########################
+
+class Benchmarking
+  ROOT = Pathname File.expand_path(['..', '..'].join(File::Separator), __FILE__)
+  TMP_DIR_BASE = File.join('tmp', 'bench')
+  TMP_DIR = File.join(ROOT, TMP_DIR_BASE)
+
+  attr_accessor :url_base
+
+  def initialize
+    @url_base = "http://localhost:9292"
+    refresh_temp_dir
```

Comment:
all the code required to run the benchmark is copied to and run from a temp directory so that it can be run across revisions. otherwise, it might not exist in some revisions or be different